tests/playwright/src/specs/compose-onboarding-smoke.spec.ts

```diff
@@ -66,14 +69,13 @@ describe.skipIf(isCI && isLinux)('Compose onboarding workflow verification', asy
   });
 
   test('Compose onboarding button Setup is available', async () => {
-    await navBar.openSettings();
-    const settingsBar = new SettingsBar(page);
-    const resourcesPage = await settingsBar.openTabPage(ResourcesPage);
+    settingsBar = await navBar.openSettings();
+    await settingsBar.openTabPage(ResourcesPage);
+    composeResourceCardPage = new ResourceCardPage(page, 'Compose');
```

Comment:
If you will move the `resourceCardIsVisible` then the check will happen before you actually instantiate the Card Page class. which is more logical.